plugins/PrivacyManager/SystemSettings.php

```diff
@@ -53,12 +61,35 @@ private function createTermsAndConditionUrlSetting()
         });
     }
 
-    private function createShowInEmbeddedWidgetsSetting()
+    private function createShowInEmbeddedWidgetsSetting(): SystemSetting
     {
         return $this->makeSetting('showInEmbeddedWidgets', $default = false, FieldConfig::TYPE_BOOL, function (FieldConfig $field) {
             $field->title = Piwik::translate('PrivacyManager_ShowInEmbeddedWidgets');
             $field->uiControl = FieldConfig::UI_CONTROL_CHECKBOX;
             $field->description = Piwik::translate('PrivacyManager_ShowInEmbeddedWidgetsDescription');
         });
     }
+
+    private function createForceCookielessTrackingSetting(): SystemConfigSetting
+    {
+        return $this->makeSettingManagedInConfigOnly('Tracker', 'force_cookieless_tracking', $default = false, FieldConfig::TYPE_BOOL, function (FieldConfig $field) {
```

Comment:
@sgiehl can we make this a regular setting? I don't think it needs to be managed in the config specifically. Any setting can be configured in the config anyway eg in this case it would be `[PrivacyManager]force_cookieless_tracking=0/1`.

By using a normal setting, the advantage is that when configured in the config then it won't appear in the UI anymore and it cannot be overwritten. Thus it is quite enforced. 

I think we would use `makeSettingManagedInConfigOnly ` only for settings where they were initially configured in the config and then moved into the UI for BC.

Reply:
I had implemented that as a regular setting. But then a lot tests started failing. Seems updating the tracker js is triggered in various cases where no db connection is yet configured. Checking if the `tracker.js` should be included using a SystemSetting then throws an error that even a try catch couldn't catch 🤷 
Not sure if that only happens in tests or might happen somewhen else as well.

Reply:
Not sure where it fails so bit hard to say. Generally be good though if we didn't have this stored in the config so people can disable/remove this setting from the UI etc.Not sure though how much work it be to fix it. Is it like that there are unit tests that are actually integration tests maybe and that's why it fails? Not sure where we generate tracker.js in tests.

Reply:
It is triggered whenever a plugin is activated/deactivated and on some other events. Can revert that to a system setting and try fixing the tests again. But not sure how much effort that will be. Was quicker to use the config instead. But will give it a try tomorrow

Reply:
👍

Reply:
It took me a bit to figure out what the problem is. Seems `Piwik\Settings\Storage\Factory` was created in an early state where no database connection was yet set up, causing exceptions when it was used later. Recreating the factory in StaticContainer solves that.